tests/integration/test_astype.py

```diff
@@ -21,12 +21,51 @@
 TEST_VECTOR = [0, 0, 1, 2, 3, 0, 1, 2, 3]
 ALL_BUT_COMPLEX = ["?", "b", "h", "i", "l", "B", "H", "I", "L", "e", "f", "d"]
 ALL_TYPES = ALL_BUT_COMPLEX + ["F", "D"]
+ORDER = ("C", "F", "A", "K")
+CASTING = ("no", "equiv", "safe", "same_kind")
+UNSIGNED_TYPE = ["b", "h", "i", "l"]
+SIGNED_TYPE = ["B", "H", "I", "L"]
+ALL_TYPES_BUT_BOOL = [
+    "b",
+    "h",
+    "i",
+    "l",
+    "B",
+    "H",
+    "I",
+    "L",
+    "e",
+    "f",
+    "d",
+    "F",
+    "D",
+]
```

Comment:
@xialu00 I think this PR needs a rebase or merge, these same changes already went in another PR:

https://github.com/nv-legate/cunumeric/commit/7a0e7f784175e8e73c5845759d7ecd84a4b908b4

I will wait to review further until the branch is up to date.

Reply:
I update the branch up to date. Thank you for reviewing.